xds/src/main/java/io/grpc/xds/client/XdsClient.java

```diff
@@ -378,6 +386,27 @@ public Map<Bootstrapper.ServerInfo, LoadReportClient> getServerLrsClientMap() {
     throw new UnsupportedOperationException();
   }
 
+  /** Callback used to report gauge metric value for resources. */
+  public interface ResourceCallback {
+    // TODO(dnvindhya): include the "authority" label once xds.authority is available.
+    void reportResourceCountGauge(long resourceCount, String cacheState, String resourceType);
+  }
+
+  /** Callback used to report a gauge metric value for server connections. */
+  public interface ServerConnectionCallback {
+    void reportServerConnectionGauge(boolean isConnected, String xdsServer);
+  }
+
+  /**
+   * Reports whether xDS client has a "working" ADS stream to xDS server.
+   * The definition of a working stream is defined in gRFC A78.
+   * @see <a href="https://github.com/grpc/proposal/blob/master/A78-grpc-metrics-wrr-pf-xds.md#xdsclient">
+   *   A78-grpc-metrics-wrr-pf-xds.md</a>
+   */
+  public SettableFuture<Void> reportServerConnections(ServerConnectionCallback callback) {
```

Comment:
The return type should be `ListenableFuture<Void>` (or `Future<Void>`, if we don't want/care about notifications). `SettableFuture` should always be an implementation detail.

Reply:
TIL, updated to return `Future<Void>` instead.